### PR TITLE
 Improve configuration handling in BatchWithPartitionData

### LINKED ISSUE #9467: [FEA] Optimize configuration handling in BatchWithPartitionData
As a follow-up for #9110, the current implementation uses the conf `spark.rapids.sql.columnSizeBytes` to get the maximum size of column set by cudf. However, it is expensive to make a full copy of SQLConf to parse only a single config option. Callers should propagate the values from an existing RapidsConf to the method `addPartitionValuesToBatch`.

https://github.com/NVIDIA/spark-rapids/blob/30e5a8f537ee0e40ba713807d1d53df7b267109e/sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala#L174-L178

Creating a separate issue for this as this change would affect many files.

### PR DESCRIPTION
**Pull request by parthosa**
Fixes #9467. 

#### Changes:
1. Create a new config  `spark.rapids.sql.columnSizeBytes` that stores maximum size of column in bytes. 
     - This aligns with other similar configurations and ensures consistency.
2. Modified all callers to the method `addPartitionValuesToBatch()` to accept `maxGpuColumnSizeBytes` as a parameter and pass it along.

#### Testing:
- Existing unit tests and integration tests have been updated for these changes.

### CODE REVIEW

File: sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuSparkScan.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ private static class MultiFileBatchReader
                          boolean queryUsesInputFile) {
       super(task.task, task.table(), task.expectedSchema(), task.isCaseSensitive(),
           task.getConfiguration(), task.getMaxBatchSizeRows(), task.getMaxBatchSizeBytes(),
-          task.getTargetBatchSizeBytes(), task.useChunkedReader(),
+          task.getTargetBatchSizeBytes(), task.getMaxBatchSizeBytes(), task.useChunkedReader(),
```
Comment (jlowe) [inline sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuSparkScan.java:285]: This is max batch size instead of max column size?

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala [hunk 2]
```diff
@@ -157,31 +156,34 @@ class BatchWithPartitionDataIterator(batchesWithPartitionData: Seq[BatchWithPart
 object BatchWithPartitionDataUtils {
   /**
    * Splits partition data (values and row counts) into smaller batches, ensuring that
-   * size of batch is less than cuDF size limit. Then, it utilizes these smaller
+   * size of batch is less than the maximum batch size. Then, it utilizes these smaller
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala:159]: Max batch size instead of column size?

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala [hunk 3]
```diff
@@ -192,19 +194,24 @@ object BatchWithPartitionDataUtils {
...
+      partitionSchema: StructType,
+      maxGpuColumnSizeBytes: Long): GpuColumnarBatchIterator = {
+    addPartitionValuesToBatch(batch, Array(batch.numRows), Array(partitionValues), partitionSchema,
+      maxGpuColumnSizeBytes)
   }
 
   /**
    * Splits partitions into smaller batches, ensuring that the batch size
-   * for each column does not exceed the maximum column size limit.
+   * for each column does not exceed the maximum batch size limit.
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala:214]: Revert this change

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala [hunk 4]
```diff
@@ -192,19 +194,24 @@ object BatchWithPartitionDataUtils {
   }
 
   /**
-   * Adds a single set of partition values to all rows in a ColumnarBatch.
+   * Adds a single set of partition values to all rows in a ColumnarBatch ensuring that
+   * size of batch is less than the maximum batch size.
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala:198]: Comment states this is enforcing batch size limit but it does not.

File: sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuSparkScan.java (hunk 1 above)
Comment (parthosa) [reply sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuSparkScan.java]: Replaced with `getMaxGpuColumnSizeBytes()`

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala (hunk 2 above)
Comment (parthosa) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala]: Replaced with `maximum column size`

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala (hunk 4 above)
Comment (parthosa) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala]: Replaced it with `ensuring that size of column is less than the maximum column size`

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala (hunk 3 above)
Comment (parthosa) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala]: Reverted

**Review by tgravescs — APPROVED** [review]

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]
